### DISCUSSION TITLE
Pause on Exception

### DISCUSSION THREAD

**Discussion by chaitanyabsprip** [mfussenegger/nvim-dap#576] [Q&A]
Is there a way to not pause on unhandled exceptions? This is something that the ~vscode~ debugger allows to control. I'm curious(and require) whether that can be also done with this plugin

**Comment by mfussenegger** [comment]
You should be able to control that via the `dap.set_exception_breakpoints()` function. See `:help dap.set_exception_breakpoints()`

**Reply by jasalt** [reply]
I got this working but didn't yet get how to enable it by default during neovim startup as running this function on init gives "Cannot set exception breakpoints: No active session!". Currently using these keybindings (with php):

```
vim.keymap.set("n", "<leader>da", function()
  require("dap").set_exception_breakpoints({ "Warning", "Error", "Exception" })
end, { desc = "Stop on exceptions" }) -- TODO this one doesn't show on which-key

vim.keymap.set("n", "<leader>dA", function()
  require("dap").set_exception_breakpoints({ "Notice", "Warning", "Error", "Exception" })
end, { desc = "Stop on all" })
```
This just needs to be run every time after re/starting debug session. What would be correct method to enable it by default?

**Reply by mfussenegger** [reply]
Did you try the documented way:

```
    You can also set the default value via a `defaults.fallback` table:

>lua
        require('dap').defaults.fallback.exception_breakpoints = {'raised'}
<

    Or per config/adapter type:

>lua
        require('dap').defaults.python.exception_breakpoints = {'raised'}
```

**Reply by jasalt** [reply]
I noted this from manual and tried earlier but seems I had done a typo that time setting value as `{ { "Notice", "Warning", "Error", "Exception" } }`. I also didn't understand 100% if that function was what was needed so I had given up early. I tried it again now and it worked, thank you!

Added following to init.lua (for php):

```
local dap = require("dap")
dap.defaults.php.exception_breakpoints = { "Notice", "Warning", "Error", "Exception" }
```

**Comment by lucaSartore** [comment]
For everyone who is looking for an easier way to use this feature, I made a simple widget that you can find here: 
[https://github.com/lucaSartore/nvim-dap-exception-breakpoints](https://github.com/lucaSartore/nvim-dap-exception-breakpoints)
